test/extended/util/client.go

```diff
@@ -123,6 +131,15 @@ func NewCLIWithoutNamespace(project string) *CLI {
 	return cli
 }
 
+// NewCLIWithoutNamespacaInside initializes the CLI and Kube framework helpers
+// without a namespace. It can be called inside of a Ginkgo .It()
+// function. Use SetupProject() to create a project for this namespace.
+func NewCLIWithoutNamespaceInside(project string) *CLI {
```

Comment:
You can't create frameworks inside of a function, that's not what frameworks are for.  You should never have more than one framework per test

Reply:
Does this address the concern?

`
func NewCLIWithFramework(kubeFramework *framework.Framework) *CLI {
        cli := &CLI{
                kubeFramework:   kubeFramework,
                username:        "admin",
                execPath:        "oc",
                adminConfigPath: KubeConfigPath(),
        }
        return cli
}
...
// NewCLIWithoutNamespacWithFramework initializes the CLI using the provided Kube
// framework without a namespace. It can be called inside of a Ginkgo .It()
// function.
func NewCLIWithoutNamespaceWithFramework(kubeFramework *framework.Framework) *CLI {
        cli := NewCLIWithFramework(kubeFramework)
        cli.kubeFramework.SkipNamespaceCreation = true
        return cli
}`

Reply:
No, just don't use this path.  Yeah, new with framework is fine.  Don't do two variants, just straight up take a framework.